File: CSharpCodeBase/entities/items/grenade.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class grenade :entity{
 var GrenadeInventoryEntity= Class({__includes=Entity, init=function(self, entity, name)            
   Entity.init(self);
   this.name = name;
   this.owner = entity;
 }})
 public void GrenadeInventoryEntity:Apply(){
     this.owner.grenadeModeVisualizer:Disable();
     this.owner.grenadeModeVisualizer:Enable(this.name);
 }
 InventoryEntity}}

File: CSharpCodeBase/entities/items/grenade.lua
local Class = require 'utils.hump.class'
local Entity = require 'unity.entity'
local RecipeList = require 'components.inventory.recipelist'

local GrenadeInventoryEntity= Class({__includes=Entity, init=function(self, entity, name)            
  Entity.init(self)
  self.name = name
  self.owner = entity
end})

function GrenadeInventoryEntity:Apply()
    self.owner.grenadeModeVisualizer:Disable()
    self.owner.grenadeModeVisualizer:Enable(self.name)
end

return GrenadeInventoryEntity